src/renderer/components/cluster-manager/cluster-topbar.tsx

```diff
@@ -19,26 +19,33 @@
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
+import { observer } from "mobx-react";
 import React from "react";
+import type { RouteComponentProps } from "react-router";
 import { catalogURL } from "../../../common/routes";
-import type { Cluster } from "../../../main/cluster";
 import { navigate } from "../../navigation";
 import { Icon } from "../icon";
 import { TopBar } from "../layout/topbar";
 import { MaterialTooltip } from "../material-tooltip/material-tooltip";
+import type { Cluster } from "../../../main/cluster";
+import { ClusterStore } from "../../../common/cluster-store";
+import type { ClusterViewRouteParams } from "../../../common/routes";
 
-interface Props {
-  cluster: Cluster
+interface Props extends RouteComponentProps<ClusterViewRouteParams> {
 }
 
-export function ClusterTopbar({ cluster }: Props) {
+export const ClusterTopbar = observer((props: Props) => {
+  const getCluster = (): Cluster | undefined => {
+    return ClusterStore.getInstance().getById(props.match.params.clusterId);
```

Comment:
Shouldn't the props be `entityId` ?

Reply:
Props are coming automatically from the `<Route/>` wrapper.

Reply:
True, but if other topbars are registered for non-clusters, it seems weird that it would pass in the `entityId` as `clusterId`.